src/Api/PubnubApi/EventEngine/Subscribe/SubscribeEventEngineFactory.cs

```diff
@@ -6,36 +6,37 @@ namespace PubnubApi.EventEngine.Subscribe
 {
 	public class SubscribeEventEngineFactory
 	{
-		private ConcurrentDictionary<string, SubscribeEventEngine> engineinstances;
+		private ConcurrentDictionary<string, SubscribeEventEngine> engineInstances { get; set;}
 		internal SubscribeEventEngineFactory()
-		{
-			this.engineinstances = new ConcurrentDictionary<string, SubscribeEventEngine>();
+		{ 
+			this.engineInstances = new ConcurrentDictionary<string, SubscribeEventEngine>();
 		}
-		internal bool hasEventEngine(string instanceId)
+		internal bool HasEventEngine(string instanceId)
 		{
-			return engineinstances.ContainsKey(instanceId);
+			return engineInstances.ContainsKey(instanceId);
 		}
-		internal SubscribeEventEngine getEventEngine(string instanceId)
+		internal SubscribeEventEngine GetEventEngine(string instanceId)
 		{
 			SubscribeEventEngine subscribeEventEngine;
-			engineinstances.TryGetValue(instanceId, out subscribeEventEngine);
+			engineInstances.TryGetValue(instanceId, out subscribeEventEngine);
 			return subscribeEventEngine;
 		}
 
-		internal SubscribeEventEngine initializeEventEngine<T>(string instanceId,
+		internal SubscribeEventEngine InitializeEventEngine(string instanceId,
 			Pubnub pubnubInstance,
 			PNConfiguration pubnubConfiguration,
 			SubscribeManager2 subscribeManager,
 			Action<Pubnub, PNStatus> statusListener = null,
-			Action<Pubnub, PNMessageResult<T>> messageListener = null)
+			Action<Pubnub, PNMessageResult<object>> messageListener= null)
```

Comment:
Does the EE implementation still use the "cast all messages to object even though the PNMessageResult is, in theory, generic" approach?

Reply:
Earlier there used to be problems when user make multiple subscribe calls with different <T> types (may be by mistake sometimes) which can lead to potential cat exception. Using super class/object type, It can be a bit easy for serialisation utility after getting type knowledge from channelType map.

btw I got this approach from one of the old PR which we had missed in past to merge.
And I interpreted it like:

Reply:
Also It's hard to predict that T in listener callback will match to the type of message we have in results. So it seems keeping it in general type make sense here when we have to serialise it